lib/Github/Api/PullRequest.php

```diff
@@ -163,6 +164,10 @@ public function merge($username, $repository, $id, $message, $sha, $mergeMethod
             $mergeMethod = $mergeMethod ? 'squash' : 'merge';
         }
 
+        if (!in_array($mergeMethod, array('merge', 'squash', 'rebase'), true)) {
+            throw new InvalidArgumentException(sprintf('"$mergeMethod" must be one of ["merge", "squash", "rebase"] ("%s" given).', $mergeMethod));
```

Comment:
We argue about how smart/helpful or ("helpful") the client should be. 

My opinion is not more valid than any other, but will I approve this because of consistency. We throw `InvalidArugmentException` when we know the API is going to return a fail in Review::create, CurrentUser/Emails::add, Issue/Labels::add. And as mentioned before; on plenty of places we silently correct wrong input parameters.